framework/src/main/java/org/checkerframework/common/basetype/BaseTypeChecker.java

```diff
@@ -600,6 +623,17 @@ private void printStoredMessages(CompilationUnitTree unit) {
          */
         final BaseTypeChecker checker;
 
+        /** Stores the stack trace till the point the checker encounters an error. */
+        StackTraceElement[] trace;
+
+        /**
+         * Constructor method for checker message without stack trace storage.
+         *
+         * @param kind kind of diagnostic
+         * @param message error message that needs to be printed
+         * @param source tree element causing the error
+         * @param checker the type-checker in use
+         */
         private CheckerMessage(
                 Diagnostic.Kind kind, String message, Tree source, BaseTypeChecker checker) {
             this.kind = kind;
```

Comment:
This constructor should call the other constructor, providing `null` for the trace.

Reply:
I have made the changes in my latest commit